lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -424,6 +433,39 @@ def _user_certificate_info(self, user, context, course_grade, bulk_certs):
         )
         return certificate_info
 
+    @staticmethod
+    def _generate_user_info(user):
+        """
+        Returns the user and user profile information for the given user.
+        """
+
+        profile = user.profile
+        position = profile.position
+        result_list = [
+            user.id,
+            user.email,
+            user.username,
+            user.first_name,
+            profile.second_name,
+            user.last_name,
+            user.profile.additional_email or '',
+            profile.phone or '',
+            profile.date_of_birth or '',
+            profile.region_display or '',
+        ]
+
+        if profile.position is not None:
+            has_parent = position.parent is not None
+            result_list.extend([
+                position.parent.name if has_parent else position.name,
+                position.name if has_parent else profile.other_position,
+                ', '.join(profile.specialization.values_list('name', flat=True)) if position.has_specialization else '',
+            ])
```

Comment:
I would extend result with three empty strings in else condition and handle only one return in the function.